Core/GOAP/GoapAgentState.cs

```diff
@@ -9,4 +11,5 @@ public sealed class GoapAgentState
     public int ConsumableCorpseCount { get; set; }
     public int LastCombatKillCount { get; set; }
     public bool Gathering { get; set; }
+    public LinkedList<Vector3> safeLocations = new LinkedList<Vector3>();
```

Comment:
In my humble opinion, making `safeLocations` as a `LinkedList<Vector3>` seems like an overkill. 

In my humble opinion, a safe spot would be a location what the player was already stepped on in the **recent past**. 

While the player is outside of combat, periodically `X` seconds or `X` milliseconds, a location can be marked as safe.

Upon entering the `FleeGoal` the goal execution would be really simple, from a `Stack` collection just pops safe locations until leaves combat.

There should be some guard clause that the `safeLocations` should contain about `Y` seconds worth of travel distance in order to make sure that do not run out of `safeLocations` this can be checked in the `CanRun()`. 

Basically `safeLocations.Count > 0`